File: leetcode/leetcode-patterns/contains-duplicate.py
class Solution(object):
  def containsDuplicate(self, nums):
    """
    :type nums: List[int]
    :rtype: bool
    """
    return len(set(nums)) < len(nums)

  def containsDuplicate2(self, nums):
    """
    :type nums: List[int]
    :rtype: bool
    """
    s = set()
    for n in nums:
      if n not in s:
        s.add(n)
      else:
        return True
    return False
        
        
        
        

File: leetcode/top-interview-questions/easy/array/contains-duplicate.rs
use std::collections::HashSet;

impl Solution {
    pub fn contains_duplicate(nums: Vec<i32>) -> bool {
        let mut num_set = HashSet::new();
        for num in nums {
            if !num_set.insert(num) {
                return true;
            }
        }

        return false;
    }
} 